src/modules/metadata.js

```diff
@@ -70,10 +74,18 @@ export const getProgramAsMetadata = (program) => ({
     [program.id]: program,
 })
 
-export const getDynamicTimeDimensionsMetadata = (program, stage) => ({
+export const getDynamicTimeDimensionsMetadata = (
+    program,
+    stage,
+    inputType
+) => ({
     ...Object.values(getTimeDimensions()).reduce((acc, dimension) => {
-        acc[dimension.id] = {
-            id: dimension.id,
+        const dimensionId = getTimeDimensionMetadataId(
```

Comment:
It might be possible to replace `getTimeDimensionMetadataId` with the existing `formatDimensionId` like:

`formatDimensionId({ dimensionId: dimension.id, programId: program?.id, outputType: inputType })`